src/eliminate_identity.cpp

```diff
@@ -0,0 +1,35 @@
+#include <migraphx/eliminate_identity.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/operators.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/stringutils.hpp>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+void eliminate_identity::apply(program& p) const
+{
+    for(auto ins : iterator_for(p))
+    {
+        if(ins->get_operator().name() == "identity")
+        {
+            if(ins != p.end())
+            {
+                instruction_ref identity_input{ins->inputs().at(0)};
+                auto next_ins = std::next(ins);
+                std::vector<instruction_ref> next_ins_inputs{next_ins->inputs()};
+                std::replace_if(next_ins_inputs.begin(),
+                                next_ins_inputs.end(),
+                                [&](instruction_ref& input) { return input == ins; },
+                                identity_input);
+                p.replace_instruction(next_ins, next_ins->get_operator(), next_ins_inputs);
+            }
```

Comment:
I don't think ins->outputs() will work for identity since this value will only be one. However, I might be able to iterate through each instruction and check if the inputs come from an identity.

Reply:
> Can you check the output of the identity instruction, then replace the inputs of the identity's output with the identity's input?

Thats what `replace_instruction(ins, rep)` will already do.

Reply:
> I don't think ins->outputs() will work for identity since this value will only be one. However, I might be able to iterate through each instruction and check if the inputs come from an identity.

The `ins->outputs()` will be all the instructions that take the instruction as input. So if the instruction is used in multiple places it will be greater than 1.

Reply:
I had changed my logic before seeing this. If I can simply further, I will do so.

Reply:
And if the instruction is not used anywhere than `ins->outputs().size()` will be zero.